src/Select.js

```diff
@@ -177,7 +177,7 @@ var Select = React.createClass({
 					newProps.placeholder)
 				);
 			};
-			if (this.props.asyncOptions) {
+			if (this.props.asyncOptions && this.props.autoload) {
```

Comment:
I don't think so, the use case is that the component renders with a null value and then a RESTful promise returns to populate the value of the react-select as an object with a label and a value.  With the object, react-select has enough information to display the correct value without polling the `this.props.asyncOptions` function.

In the proposed case of checking for `newProps.value`, I think it is only beneficial to call `this.props.asyncOptions` if it is a string since it won't have the label to display.

So if `this.props.asyncOptions` is specified and `newProps.value` is a string then it should always call `this.props.asyncOptions` to get the full label/value object.  If `newProps.value` is an object then it should only call `this.props.asyncOptions` if `this.props.autoload` is true.